### PR TITLE
Add donut CPU inference ORT 

### PR DESCRIPTION
**Pull request by mht-sharma**
# What does this PR do?

Add donut inference using ONNXRuntime on cpu.
#743 

IOBinding support will be added separately for complex models as discussed internally.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/exporters/onnx/config.py [hunk 1]
```diff
@@ -338,6 +338,10 @@ def __init__(
             self._encoder_onnx_config = encoder_onnx_config_constructor(config.encoder)
             self._normalized_config.ENCODER_NORMALIZED_CONFIG_CLASS = self._encoder_onnx_config._normalized_config
 
+            self.seq_len_scale = 1
```
Comment (fxmarty) [inline optimum/exporters/onnx/config.py:341]: What does `seq_len_scale` stand for? There is no occurrence of it in transformers code base.

File: optimum/exporters/onnx/config.py [hunk 2]
```diff
@@ -406,4 +410,27 @@ def outputs(self) -> Dict[str, Dict[int, str]]:
         # make sure this output is moved at the end.
         if "encoder_last_hidden_state" in common_outputs:
             common_outputs.move_to_end("encoder_last_hidden_state")
+
+        if self._behavior is ConfigBehavior.ENCODER:
+            common_outputs["last_hidden_state"][1] = f"{common_outputs['last_hidden_state'][1]} / {self.seq_len_scale}"
```
Comment (fxmarty) [inline optimum/exporters/onnx/config.py:415]: This renames dynamic axis 1 as `XXX / 1` in most cases, right?

File: optimum/exporters/onnx/model_configs.py [hunk 3]
```diff
@@ -565,7 +565,9 @@ class MobileNetV2OnnxConfig(MobileNetV1OnnxConfig):
 
 
 class DonutSwinOnnxConfig(ViTOnnxConfig):
-    pass
+    def __init__(self, config: "PretrainedConfig", task: str = "default"):
+        super().__init__(config, task)
+        self.seq_len_scale = (self._config.patch_size**2) * (4 ** (len(self._config.depths) - 1))
```
Comment (fxmarty) [inline optimum/exporters/onnx/model_configs.py:570]: Why is it needed?

**Review by fxmarty — COMMENTED** [review]
Thank you! Could you make sure that the tests

```
RUN_SLOW=1 pytest tests/exporters/onnx/test_onnx_export.py -k "test_pytorch_export and donut-swin"
RUN_SLOW=1 pytest tests/exporters/onnx/test_onnx_export.py -k "test_pytorch_export and donut_swin"
```

pass as well?

**Comment by fxmarty** [comment]
@mht-sharma Actually, I am not sure to understand what this PR does. `donut-swin` is already supported it appears: https://github.com/huggingface/optimum/blob/dbb43fb622727f2206fa2a2b3b479f6efe82945b/tests/exporters/exporters_utils.py#L49 & https://github.com/huggingface/optimum/blob/dbb43fb622727f2206fa2a2b3b479f6efe82945b/optimum/exporters/tasks.py#L358-L360

File: tests/exporters/exporters_utils.py [hunk 4]
```diff
@@ -121,8 +121,7 @@
         ],
         "microsoft/trocr-small-handwritten": ["vision2seq-lm"],
     },
-    # Disabled for now because of high atol threshold requirement to pass
-    # "donut-swin": "hf-internal-testing/tiny-random-DonutSwinModel"
+    "donut-swin": "hf-internal-testing/tiny-random-DonutSwinModel",
```
Comment (fxmarty) [inline tests/exporters/exporters_utils.py:124]: This is already in the dictionary

**Comment by mht-sharma** [comment]
> @mht-sharma Actually, I am not sure to understand what this PR does. `donut-swin` is already supported it appears:
> 
> https://github.com/huggingface/optimum/blob/dbb43fb622727f2206fa2a2b3b479f6efe82945b/tests/exporters/exporters_utils.py#L49
> 
> &
> https://github.com/huggingface/optimum/blob/dbb43fb622727f2206fa2a2b3b479f6efe82945b/optimum/exporters/tasks.py#L358-L360

There are few exports which were broken, the earlier PR just export `donut-swin` encoder, but the vision-encoder-decoder export was not tested. Plus IOBinding support will also be added.

**Review by fxmarty — APPROVED** [review]
Thank you!

**Pull request merged**